draft-ietf-tls-tls13.md

```diff
@@ -4319,6 +4322,25 @@ will receive various TLS 1.x versions in this field, however its value
 MUST always be ignored.
 
 
+## Zero-RTT backwards compatibility
+
+A TLS 1.3 ClientHello with 0-RTT data is not compatible with older servers.
+This can cause compatibility issues if a client attempts to use 0-RTT with a
+server cluster that contains servers without TLS 1.3 support. For example, a
+multi-server deployment may deploy TLS 1.3 gradually with some servers
+implementing TLS 1.3 and some implementing TLS 1.2. This may also occur if a
+TLS 1.3 deployment gets reverted to TLS 1.2.
+
+If a TLS 1.3 client sends a ClientHello with 0-RTT to an older server, the
+server will respond with a ServerHello containing an older version number. The
+handshake will later fail at the 0-RTT data. Clients which accept older
+versions of TLS MAY retry the connection without 0-RTT in response to this
+ServerHello. It is NOT RECOMMENDED to retry the connection in response to a
+more generic error or advertise lower versions of TLS.
```

Comment:
I'm not sure I follow. Do you mean security properties of the 0-RTT retry or of advertising lower versions of TLS? I'm saying not to do that and to retry at TLS 1.3. (Of course, all us browsers will certainly do a version fallback too. It seems 1.3 intolerance is rampant. But that's another matter and, yes, we have the server_random signaling and whatnot to help there.)

I don't know of any particular security considerations for this retry. I guess it's another vector for 0-RTT data to be replayable, but we're already accepting that it is, right? (Hrm. Actually... this would defeat any server-side stateful 0-RTT replay protection... I'm dubious anyone would bother with that, but still.)